packages/starlight-links-validator/libs/validation.ts

```diff
@@ -216,6 +216,7 @@ function validateSelfHash(context: ValidationContext) {
 
   const hash = link.raw.split('#')[1] ?? link.raw
   const sanitizedHash = hash.replace(/^#/, '')
+  if (sanitizedHash == '_top') return
```

Comment:
```suggestion
  if (sanitizedHash === '_top') return
```

While this needs fixing, not sure yet if this line is the proper approach to the issue and would stay here in the end.